src/slurm_plugin/common.py

```diff
@@ -244,7 +245,13 @@ def _update_dns_hostnames(self, nodes):
             # Submit calls to change_resource_record_sets in batches of 500 elements each.
             # change_resource_record_sets API call has limit of 1000 changes,
             # but the UPSERT action counts for 2 calls
-            route53_client = boto3.client("route53", region_name=self._region, config=self._boto3_config)
+            # Also pick the number of retries to be the max between the globally configured one and 3.
+            # This is done to address Route53 API throttling without changing the configured retries for all API calls.
+            configured_retry = self._boto3_config.retries.get("max_attempts", 0) if self._boto3_config.retries else 0
+            boto3_config = self._boto3_config.merge(
```

Comment:
What about moving that logic in a common function?  We're overriding `boto3_config` to be sure to have a max attempts greater than "a specified value" and the code is very specific and strictly related to the `boto3_config.retries` structure.